File: shared/types/inventory.ts
/**
 * Shared inventory types
 */

export interface Inventory {
  item_id: string;
  current_stock: number;
  reorder_point: number;
  safety_stock: number;
  lead_time: number;
  days_to_rupture: number;
  alert_level: 'normal' | 'medium' | 'high' | 'critical';
}

export interface ReorderPointCalculation {
  average_demand: number;
  lead_time_demand: number;
  safety_stock: number;
  reorder_point: number;
  service_level: number;
}

File: backend/app/api/v1/schemas/inventory.py
"""
Inventory schemas
"""
from pydantic import BaseModel
from typing import Literal


class InventoryResponse(BaseModel):
    """Inventory response"""
    item_id: str
    current_stock: int
    reorder_point: int
    safety_stock: int
    days_to_rupture: int
    alert_level: Literal["normal", "medium", "high", "critical"]


class ReorderPointCalculation(BaseModel):
    """Reorder point calculation result"""
    average_demand: float
    lead_time_demand: float
    safety_stock: float
    reorder_point: float
    service_level: float

